osclib/stagingapi.py

```diff
@@ -356,20 +356,25 @@ def supseded_request(self, request):
             msg = msg.format(request_id, action)
             logging.info(msg)
 
+        pkg_do_supersede = True
+        if len(target_pkgs):
+            if target_package not in target_pkgs:
+                pkg_do_supersede = False
+
         # If the package is currently tracked then we do the replacement
         stage_info = self.packages_staged.get(target_package, {'prj': '', 'rq_id': 0})
-        if stage_info['rq_id'] != 0 and int(stage_info['rq_id']) != request_id:
+        if  pkg_do_supersede is True and stage_info['rq_id'] != 0 and int(stage_info['rq_id']) != request_id:
```

Comment:
I find this line suspicious, `stage_info['rq_id']` looks like that can have strings (converted to int()) or 0 value.  I would suggest this:

`if pkg_do_supersede and stage_info['rq_id'] and int(stage_info['rq_id']) != request_id:`

A better approach is to move `stage_info['rq_id'] and int(stage_info['rq_id']) != request_id` into another helper that try to convert into a int and compare.